mypy error in groupbyclass.py

Fix the error causing this:
```
Run python3 -m mypy arkouda
arkouda/groupbyclass.py:338: error: Incompatible types in assignment (expression has type "Any | None", variable has type "pdarray | Strings | Categorical | tuple[pdarray | Strings | Categorical, ...]")  [assignment]
arkouda/groupbyclass.py:339: error: Incompatible types in assignment (expression has type "Any | None", variable has type "pdarray")  [assignment]
arkouda/groupbyclass.py:340: error: Incompatible types in assignment (expression has type "Any | None", variable has type "pdarray")  [assignment]
arkouda/groupbyclass.py:352: error: Incompatible types in assignment (expression has type "Any | None", variable has type "pdarray")  [assignment]
arkouda/groupbyclass.py:353: error: Incompatible types in assignment (expression has type "Any | None", variable has type "pdarray")  [assignment]
arkouda/numpy/util.py:898: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
arkouda/numpy/segarray.py:6[7](https://github.com/Bears-R-Us/arkouda/actions/runs/15326421515/job/43121985007#step:6:8): note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Found 5 errors in 1 file (checked 101 source files)
```

--- a/arkouda/groupbyclass.py
+++ b/arkouda/groupbyclass.py
@@ -335,9 +335,12 @@ class GroupBy:
         ):
             self.keys = cast(groupable, kwargs.get("orig_keys", None))
             drop_na_keys()
-            self.unique_keys = kwargs.get("unique_keys", None)
-            self.permutation = kwargs.get("permutation", None)
-            self.segments = kwargs.get("segments", None)
+            self.unique_keys = cast(
+                Union[pdarray, Strings, Categorical, Tuple[Union[pdarray, Strings, Categorical], ...]],
+                kwargs["unique_keys"],
+            )
+            self.permutation = cast(pdarray, kwargs["permutation"])
+            self.segments = cast(pdarray, kwargs["segments"])
             self.nkeys = len(self.keys)
             self._uki = self.permutation[self.segments]
         elif (
@@ -349,8 +352,8 @@ class GroupBy:
             self.keys = cast(groupable, kwargs.get("orig_keys", None))
             drop_na_keys()
             self._uki = kwargs.get("uki", None)
-            self.permutation = kwargs.get("permutation", None)
-            self.segments = kwargs.get("segments", None)
+            self.permutation = cast(pdarray, kwargs["permutation"])
+            self.segments = cast(pdarray, kwargs["segments"])
             self.nkeys = len(self.keys) if isinstance(self.keys, Sequence) else 1
             if not isinstance(self.keys, Sequence):
                 self.unique_keys = self.keys[self._uki]

